Refact: Use of markup protocols in case if and unless

ingSamuelG opened the pull request.

ingsissam committed:
Refact: Use of markup protocols in case if and unless

ingsissam committed:
fix : wrong name fo r the bolck unless

Spadavecchia commented on .gitignore:
```diff
@@ -6,3 +6,4 @@
 erl_crash.dump
 /cover
 /bench/results/
+*.exs#
```
This is not necessary. Delete this files locally, do not add it to .gitignore

Spadavecchia commented on lib/liquid/combinators/lexical_token.ex:
```diff
@@ -109,9 +109,11 @@ defmodule Liquid.Combinators.LexicalToken do
+    parsec(:ignore_whitespaces)
     |> choice([string("nil"), string("null"), string("NIL"), string("NULL")])
     |> replace(nil)
+
```
Comments  are not for leave dead code. Use comments only to document or explain code.

Spadavecchia commented on lib/liquid/combinators/tags/case.ex:
```diff
@@ -1,24 +1,4 @@
 defmodule Liquid.Combinators.Tags.Case do
-  @moduledoc """
```
What happened with moduledoc?

Spadavecchia commented on lib/liquid/combinators/tags/case.ex:
```diff
@@ -48,8 +27,9 @@ defmodule Liquid.Combinators.Tags.Case do
     combinator
     |> choice([
       parsec(:condition),
+      # TO-DO: nill to string is "" and the potocolls does by default 
```
TODO is for describe a future task. What is the task described here?

Spadavecchia commented on test/tags/standard_tag_test.exs:
```diff
@@ -90,15 +90,15 @@ defmodule StandardTagTest do
   end
 
   # TODO: Fix this tests for translator / Open: new parser's vars do not diferenciate between single and double quote as requeriment
```
TODO must be deleted after finished the task

Spadavecchia commented on lib/liquid/combinators/tags/if.ex:
```diff
@@ -68,6 +68,6 @@ defmodule Liquid.Combinators.Tags.If do
     ])
     |> optional(times(parsec(:logical_condition), min: 1))
-    |> tag(:if_condition)
+    |> tag(:evaluation)
```
`evaluation` is too much generic. An evaluation can be anything. A better name can be `boolean_condition` or simply `condition` https://en.wikipedia.org/wiki/Conditional_(computer_programming)

ingsissam committed:
fix: to match the merge requirements

ingsissam committed:
fix: typo

Spadavecchia committed:
fix: order code

Follow the elixir style guide: https://github.com/christopheradams/elixir_style_guide#module-attribute-ordering
